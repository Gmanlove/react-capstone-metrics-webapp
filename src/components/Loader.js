import React from 'react';
import '../styles/loader.css';

const Loader = () => (
  <div className="loading">
    <div />
    <div />
    <div />
    <div />
  </div>
);

export default Loader;
