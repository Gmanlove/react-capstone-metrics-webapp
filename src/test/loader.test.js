import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Loader from '../components/Loader';

describe('Loader Component Tests', () => {
  it('should render Loader component with a snapshot', () => {
    // Arrange & Act
    const tree = renderer.create(<Loader />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });

  it('should have a "loading" class on the first child element', () => {
    // Arrange & Act
    const { container } = render(<Loader />);

    // Assert
    expect(container.firstChild.classList.contains('loading')).toBe(true);
  });

  it('should have only one element with "loading" class', () => {
    // Arrange & Act
    const { container } = render(<Loader />);

    // Assert
    expect(container.getElementsByClassName('loading').length).toBe(1);
  });
});
