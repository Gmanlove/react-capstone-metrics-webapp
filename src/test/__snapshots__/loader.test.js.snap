// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Tests should render Loader component with a snapshot 1`] = `
<div
  className="loading"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
